[addons] change audioencoder interface to C++ style

AlwinEsch opened the pull request:
<!--- Provide a general summary of your change in the Title above -->

## Description
This is the next addon type where becomes changed to new style.

The changes are nearly equal like on the others.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

AlwinEsch committed:
[settings] change audiocds.encoder to audioencoder.xbmc.builtin.aac

audioencoder.xbmc.builtin.wav is no more present in Kodi (only as external
addon)

AlwinEsch committed:
[addons] change internal audioencoder addon.xml to generate dummy lib include to match right platform

ksooo commented on xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/AudioEncoder.h:
```diff
@@ -0,0 +1,223 @@
+                       std::string year,
+                       std::string track,
+                       std::string genre,
+                       std::string comment,
```
by-value or by-reference? </code>std::string</code> vs. <code>const std::string&</code>?

ksooo commented on xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/AudioEncoder.h:
```diff
@@ -0,0 +1,223 @@
+    /// \param[in] pbtStream            the input buffer
+    /// \return                         Number of bytes consumed
+    ///
+    virtual int Encode(int numBytesRead, const uint8_t* pbtStream)=0;
```
Please add blacks around "=" => <code>) = 0;</code>

PeterPablo replied:
blanks

ksooo replied:
Haha

AlwinEsch committed:
[cmake] only add addon.xml.in as addon.xml to install_data and nothing else!

This fix a bad problem if normal install want to install a directory who was wrongly from
before as file.

FernetMenta commented on system/settings/settings.xml:
```diff
@@ -557,7 +557,7 @@
         <setting id="audiocds.encoder" type="addon" label="621" help="36287">
           <level>3</level>
-          <default>audioencoder.xbmc.builtin.wav</default>
+          <default>audioencoder.xbmc.builtin.aac</default>
```
btw: builtin encoders addons should be dropped in favour of ffmpeg

AlwinEsch committed:
[addons] change audioencoder interface to C++ style

This is the next addon type where becomes changed to new style.

The changes are nearly equal like on the others.

ksooo reviewed (approved):
Good to go, imo.

AlwinEsch committed:
[addons] update build in audio encoder description

AlwinEsch commented:
On last commit becomes the addon type name changed to "kodi.audioencoder"

@MartijnKaijser is this OK for you?

AlwinEsch committed:
change addon type name xbmc.audioencoder to kodi.audioencoder

AlwinEsch commented:
jenkins build this please

AlwinEsch commented:
jenkins build this please
